packages/react-vapor/src/components/slider/MiddleSlider.tsx

```diff
@@ -0,0 +1,133 @@
+import 'rc-slider/assets/index.css';
+
+import {Range, SliderProps} from 'rc-slider';
+import * as React from 'react';
+import {connect} from 'react-redux';
+
+import {IDispatch} from '../../utils/ReduxUtils';
+import {SliderActions} from './SliderActions';
+import SliderHandle from './SliderHandle';
+import {
+    convertInitialValuetoRangeValue,
+    getComputedRangeValue,
+    getCrossingPoint,
+    getValuesPositionOnRange,
+    handleIsAtCrossingPoint,
+    propsValidator,
+    valuesPositionOnRange,
+} from './SliderUtils';
+
+export interface MiddleSliderOwnProps extends SliderProps {
+    id: string;
+    enabled?: boolean;
+    hasTooltip?: boolean;
+    initialValue?: number;
+    marks?: {[key: number]: string};
+    range: [number, number];
+    step?: number;
+    tabIndex?: number[];
+    onChange?: (rangeOutputValue: number) => any;
+    customTooltip?: (value: any) => JSX.Element;
+}
+
+export const mapDispatchToProps = (dispatch: IDispatch, ownProps: MiddleSliderOwnProps) => ({
+    setOutputValue: (value: number) => dispatch(SliderActions.setValue(ownProps.id, value)),
+});
+
+const MiddleSliderDisconnected: React.FunctionComponent<MiddleSliderOwnProps &
+    ReturnType<typeof mapDispatchToProps>> = ({
+    enabled = true,
+    customTooltip,
+    hasTooltip,
+    initialValue,
+    marks,
+    range,
+    setOutputValue,
+    step,
+    onChange,
+}) => {
+    const crossingPoint = getCrossingPoint(range);
+    const [highRange, setHighRange] = React.useState(crossingPoint);
+    const [lowRange, setLowRange] = React.useState(crossingPoint);
+    const rangeOutputValue = getComputedRangeValue(lowRange, highRange, range, crossingPoint);
+
+    if (onChange) {
+        onChange(rangeOutputValue);
+    }
```

Comment:
Do we really want onChange to be called each time the component renders? I think it should be added as a side effect

```ts
React.useEffect(() => {
    onChange?.(rangeOutputValue);
}, [onChange, rangeOutputValue]);
```

Reply:
freaking awesome. this code feels good.